src/services/Blockchain/useStartfiContracts.ts

```diff
@@ -0,0 +1,41 @@
+import { useActiveWeb3React } from 'hooks'
+import { useStartFiNft, useStartFiToken, useStartFiMarketplace, useStartFiRoyality } from 'hooks/useContract'
+import { useEffect, useMemo } from 'react'
+import { useWalletModalToggle } from 'state/application/hooks'
+
+/**
+ *
+ * @param contractName Name of the smart contract that we are going to use
+ * @param signer Flag to decide if the transaction is going to read or write on the blockchain
+ * @returns {
+ * account: wallet address,
+ * library: provider,
+ * contract: smart contract object
+ * }
+ */
+export const useStartFiContract = (contractName: string, signer = true) => {
+  const { account, library } = useActiveWeb3React()
+  const toggleWalletModal = useWalletModalToggle()
+  const nft = useStartFiNft(signer)
+  const token = useStartFiToken(signer)
+  const marketplace = useStartFiMarketplace(signer)
+  const nftRoyality = useStartFiRoyality(signer)
+  useEffect(() => {
+    if (!account) {
+      toggleWalletModal()
+    }
+  }, [account])
+  return useMemo(() => {
+    const contract =
+      contractName === 'nft'
```

Comment:
why one file that handles all contracts? Why not a single file for each, so we can only import the data we need rather than import a big chunk of data I don't need?
@ahmedgalalmohamed2016 Kindly confirm and feel free to correct me if I'm wrong

Reply:
@EmanHerawy  Yes I think you are right, I  removed `useStartfiContracts.ts` and uploaded contract for each file. You can check [startfiNft.ts](https://github.com/StartFi/startfi-interface/blob/feature/bc-mintNFT/src/hooks/startfiNft.ts) as example